jung-api/src/test/java/edu/uci/ics/jung/graph/CTreeMutationTest.java

```diff
@@ -227,21 +225,16 @@ private int controlledNumEdgesToRemove(Random gen) {
     return tree.successors(node).isEmpty();
   }
 
-  private static <N> List<EndpointPair<N>> descendantEdges(
-      CTree<N> tree, EndpointPair<N> startingEdge) {
-    // TODO: Migrate to com.google.common.graph.Traverser when it is available.
-    TreeTraverser<EndpointPair<N>> traverser = TreeTraverser.using(edge -> outEdges(tree, edge));
-    return traverser
-        .breadthFirstTraversal(startingEdge)
-        .skip(1) // skip the starting edge itself
-        .copyInto(new ArrayList<>());
-  }
-
-  private static <N> Set<EndpointPair<N>> outEdges(CTree<N> tree, EndpointPair<N> edge) {
-    N origin = edge.target();
-    return tree.successors(origin)
-        .stream()
-        .map(successor -> EndpointPair.ordered(origin, successor))
-        .collect(toSet());
+  private static <N> CTree<N> subTree(CTree<N> tree, N subTreeRoot) {
+    MutableCTree<N> result = TreeBuilder.builder().build();
+    for (N node :
+        Iterables.skip(
+            Traverser.forTree(tree).breadthFirst(subTreeRoot),
+            // skip the subtree's root node, for it has no predecessor
```

Comment:
Good comment, but I'd put it as an end-of-line comment on the skip() call.

Reply:
Done!